src/devices/Mcp3xxx/Mcp30xx32xx.cs

```diff
@@ -0,0 +1,149 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Spi;
+
+namespace Iot.Device.Adc
+{
+    /// <summary>
+    /// Mcp30xx32xx Abstract class representing the MCP 10 and 12 bit ADC devices.
+    /// </summary>
+    public abstract class Mcp30xx32xx : Mcp3xxx
+    {
+        /// <summary>
+        /// the number of input pins on the ADC
+        /// </summary>
+        protected byte PinCount;
+
+        private byte _adcResolutionBits;
+
+        /// <summary>
+        /// Constructs Mcp30xx32xx instance
+        /// </summary>
+        /// <param name="spiDevice">Device used for SPI communication</param>
+        /// <param name="pinCount">Value representing the number of pins on the device.</param>
+        /// <param name="adcResolutionBits">The number of bits of resolution for the ADC.</param>
+        public Mcp30xx32xx(SpiDevice spiDevice, byte pinCount, byte adcResolutionBits) : base(spiDevice)
+        {
+            PinCount = pinCount;
+            _adcResolutionBits = adcResolutionBits;
+        }
+
+        /// <summary>
+        /// Checks that the channel is in range of the available channels pins and throws an exception if not.
+        /// </summary>
+        /// <param name="channel">Channel to be checked</param>
+        /// <param name="channelCount">Value representing the number of channels on the device which may vary depending on the configuration.</param>
+        protected void CheckChannelRange(int channel, int channelCount)
+        {
+            if (channel < 0 || channel > channelCount - 1)
+            {
+                throw new ArgumentOutOfRangeException($"ADC channel must be within the range 0-{channelCount - 1}.", nameof(channel));
+            }
+        }
+
+        /// <summary>
+        /// Checks that the channel is in range of the available input pins and that both pins are part of a valid pairing of
+        /// input pins.
+        /// </summary>
+        /// <param name="positiveChannel">Positive channel to be checked</param>
+        /// <param name="negativeChannel">Negative channel to be checked</param>
+        /// <param name="channelCount">Value representing the number of channels on the device which may vary depending on the configuration.
+        /// If channelCount is not set then the PinCount is used.</param>
+        protected void CheckChannelPairing(int positiveChannel, int negativeChannel, int channelCount = -1)
+        {
+            CheckChannelRange(positiveChannel, (channelCount == -1) ? PinCount : channelCount);
+            CheckChannelRange(negativeChannel, (channelCount == -1) ? PinCount : channelCount);
+
+            // Check that the channel pins are adjacent and part of the same differential pairing
+            if (positiveChannel / 2 != negativeChannel / 2 || positiveChannel == negativeChannel)
```

Comment:
please explain the division (or create method to make it simpler to reason about)